components/payments/internal/storage/payments_test.go

```diff
@@ -0,0 +1,898 @@
+	upsertAccounts(t, ctx, store, defaultAccounts)
+	upsertPayments(t, ctx, store, defaultPayments)
+
+	t.Run("upsert with unknown connector", func(t *testing.T) {
+		connector := models.ConnectorID{
+			Reference: uuid.New(),
+			Provider:  "unknown",
+		}
+		p := defaultPayments[0]
+		p.ID = models.PaymentID{
+			PaymentReference: models.PaymentReference{
+				Reference: "test4",
+				Type:      models.PAYMENT_TYPE_PAYOUT,
+			},
+			ConnectorID: connector,
+		}
+		p.ConnectorID = connector
+
+		err := store.PaymentsUpsert(ctx, []models.Payment{p})
+		require.Error(t, err)
+	})
+
+	t.Run("upsert with same id", func(t *testing.T) {
+		p := defaultPayments[2]
+		p.Amount = big.NewInt(200)
+		p.Scheme = models.PAYMENT_SCHEME_A2A
+		upsertPayments(t, ctx, store, []models.Payment{p})
+
+		// should not have changed
+		actual, err := store.PaymentsGet(ctx, p.ID)
+		require.NoError(t, err)
+
+		comparePayments(t, defaultPayments[2], *actual)
+	})
+
+	t.Run("upsert with different adjustments", func(t *testing.T) {
+		p := models.Payment{
+			ID:              pid3,
+			ConnectorID:     defaultConnector.ID,
+			Reference:       "test3",
+			CreatedAt:       now.Add(-55 * time.Minute).UTC().Time,
+			Type:            models.PAYMENT_TYPE_PAYOUT,
+			InitialAmount:   big.NewInt(300),
+			Amount:          big.NewInt(300),
+			Asset:           "DKK/2",
+			Scheme:          models.PAYMENT_SCHEME_A2A,
+			SourceAccountID: &defaultAccounts[1].ID,
+			Adjustments: []models.PaymentAdjustment{
+				{
+					ID: models.PaymentAdjustmentID{
+						PaymentID: pid3,
+						CreatedAt: now.Add(-45 * time.Minute).UTC().Time,
+						Status:    models.PAYMENT_STATUS_SUCCEEDED,
+					},
+					PaymentID: pid3,
+					CreatedAt: now.Add(-45 * time.Minute).UTC().Time,
+					Status:    models.PAYMENT_STATUS_SUCCEEDED,
+					Amount:    big.NewInt(300),
+					Asset:     pointer.For("DKK/2"),
+					Metadata:  map[string]string{},
+					Raw:       []byte(`{}`),
+				},
+				{
+					ID: models.PaymentAdjustmentID{
+						PaymentID: pid3,
+						CreatedAt: now.Add(-55 * time.Minute).UTC().Time,
+						Status:    models.PAYMENT_STATUS_PENDING,
+					},
+					PaymentID: pid3,
+					CreatedAt: now.Add(-55 * time.Minute).UTC().Time,
+					Status:    models.PAYMENT_STATUS_PENDING,
+					Amount:    big.NewInt(300),
+					Asset:     pointer.For("DKK/2"),
+					Raw:       []byte(`{}`),
+				},
+			},
+		}
+
+		upsertPayments(t, ctx, store, []models.Payment{p})
+
+		actual, err := store.PaymentsGet(ctx, p.ID)
+		require.NoError(t, err)
+		comparePayments(t, p, *actual)
+	})
+
+	t.Run("upsert with refund", func(t *testing.T) {
+		p := models.Payment{
+			ID:            pID1,
+			ConnectorID:   defaultConnector.ID,
+			Reference:     "test1",
+			CreatedAt:     now.Add(-20 * time.Minute).UTC().Time,
+			Type:          models.PAYMENT_TYPE_TRANSFER,
+			InitialAmount: big.NewInt(100),
+			Amount:        big.NewInt(100),
+			Asset:         "USD/2",
+			Scheme:        models.PAYMENT_SCHEME_OTHER,
```

Comment:
Is this test data realistic? I was under the impression that since the PSPpayment contains only the data from the adjustment, this base payment object will contain the same data (payment type, amount, etc) as the adjustment that is being appended.